File: C++/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> prev;
        for(int i=0;i<nums.size();i++){
            if (prev.find(target-nums[i])!=prev.end()){
                return {prev[target-nums[i]], i};
            }else{
                prev[nums[i]]=i;
            }
        }
        return {};
    }
};

File: Go/0001-two-sum.go
func twoSum(nums []int, target int) []int {
	prev := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if idx, found := prev[complement]; found {
			return []int{idx, i}
		}
		prev[num] = i
	}
	return []int{}
}